R/pkg/R/DataFrame.R

```diff
@@ -1187,6 +1187,65 @@ setMethod("dapply",
             dataFrame(sdf)
           })
 
+#' gapply
+#'
+#' Apply a function to each group of a DataFrame.
+#'
+#' @param x A SparkDataFrame
+#' @param func A function to be applied to each group partition specified by grouping
+#'        column of the SparkDataFrame.
+#'        The output of func is a data.frame.
+#' @param schema The schema of the resulting DataFrame after the function is applied.
```

Comment:
Could you explain the differences among `data.frame`, `SparkDataFrame` and `DataFrame`